### PR TITLE
feat: add search parameters in GET /v1/logs 

### LINKED ISSUE #236: Add search parameters to GET /logs
Add a `search` parameters to GET /logs, so consumers can search for errors or repos in the logs.

### PR DESCRIPTION
**Pull request by k124k3n**
### User can find related `message` field in the log by including `search` parameters in `GET /v1/logs`

Example:

> GET /v1/logs/?search=logmessage

closes #236

### CODE REVIEW

**Review by bfabio — COMMENTED** [review]
Thanks!

Can you please also add the new parameter to [the OpenAPI spec](https://github.com/italia/developers-italia-api/blob/main/developers-italia.oas.yaml)

File: main_test.go [hunk 1]
```diff
@@ -3085,6 +3085,18 @@ func TestLogsEndpoints(t *testing.T) {
 				assert.Equal(t, "invalid date time format (RFC 3339 needed)", response["detail"])
 			},
 		},
+		{
+			description: `GET with "search" query param`,
+			query:       "GET /v1/logs?search=log%20m",
```
[the remark this answers is not in the document]
Comment (k124k3n) [reply main_test.go:3090]: Okay.

**Comment by k124k3n** [comment]
> Thanks!
> 
> Can you please also add the new parameter to [the OpenAPI spec](https://github.com/italia/developers-italia-api/blob/main/developers-italia.oas.yaml)?

Sure.

**Comment by k124k3n** [comment]
Can we improve this further by making the `search` result case insensitive?

**Comment by bfabio** [comment]
> Can we improve this further by making the `search` result case insensitive?

Yes, good idea

**Comment by k124k3n** [comment]
Why spectral didn't pass when it encounters `debug` word?

**Comment by bfabio** [comment]
> Why spectral didn't pass when it encounters `debug` word?

I'm not sure, maybe it matched some rule.

File: .gitignore [hunk 2]
```diff
@@ -10,3 +12,5 @@ test.db-wal
 
 # Helm chart package
 /developers-italia-api-*.tgz
+
+
```
Comment (bfabio) [inline .gitignore:16]:
```suggestion
```

File: developers-italia.oas.yaml [hunk 3]
```diff
@@ -643,6 +643,13 @@ paths:
           in: query
           name: to
           description: Only logs before this time (RFC 3339 datetime)
+        - schema:
+            type: string
+            maxLength: 255
+            example: 'log'
```
Comment (bfabio) [inline developers-italia.oas.yaml:649]: I'd use something more practical here

```suggestion
            example: 'BAD publiccode'
```

File: developers-italia.oas.yaml [hunk 4]
```diff
@@ -643,6 +643,13 @@ paths:
...
+          in: query
+          name: search
+          description: Only logs message like this parameter
```
Comment (bfabio) [inline developers-italia.oas.yaml:652]:
```suggestion
          description: Only logs matching the search string in their message
```

File: test/testdata/fixtures/logs.yml [hunk 5]
```diff
@@ -110,3 +110,28 @@
   created_at: 2010-07-30 23:59:59
   updated_at: 2010-12-31 23:59:59
   message: A log message
+
+- id: 06666ecc-042e-11ed-83b0-d8bbc146d165
+  created_at: 2010-08-01 23:59:59+00:00
+  updated_at: 2010-12-31 23:59:59+00:00
+  message: "DEBUG - [main] BulkOpsClient.main(): List of all configurations loaded: {numofthreads=1, impstatchkinterval=30, maxloginattempts=1}"
```
Comment (bfabio) [inline test/testdata/fixtures/logs.yml:117]: I'd use messages more in line with the context:

```suggestion
  message: "[github.com/foobar/baz] BAD publiccode.yml"
```

File: test/testdata/fixtures/logs.yml [hunk 6]
```diff
@@ -110,3 +110,28 @@
...
+
+- id: 37f904ea-042e-11ed-8176-d8bbc146d165
+  created_at: 2010-08-01 23:59:59+00:00
+  updated_at: 2010-12-31 23:59:59+00:00
+  message: "DEBUG - [main] BulkOpsClient.main(): List of all options loaded: {password=*********, clientloglevel=detailed, waitforcompletion=False, datetimeformat=usa, importloglevel=errors}"
```
Comment (bfabio) [inline test/testdata/fixtures/logs.yml:122]:
```suggestion
  message: "[github.com/foobar/baz2] BAD publiccode.yml"
```

File: test/testdata/fixtures/logs.yml [hunk 7]
```diff
@@ -110,3 +110,28 @@
...
+
+- id: 4b4a1b9c-042e-11ed-82a8-d8bbc146d165
+  created_at: 2010-08-01 23:59:59+00:00
+  updated_at: 2010-12-31 23:59:59+00:00
+  message: "DEBUG - [main] BulkOpsClient.submitImportRequest(): Execution complete."
```
Comment (bfabio) [inline test/testdata/fixtures/logs.yml:127]:
```suggestion
  message: "[github.com/foobar/baz3] BAD publiccode.yml"
```

File: test/testdata/fixtures/logs.yml [hunk 8]
```diff
@@ -110,3 +110,28 @@
...
+
+- id: 5332f73e-042e-11ed-8e74-d8bbc146d165
+  created_at: 2010-08-01 23:59:59+00:00
+  updated_at: 2010-12-31 23:59:59+00:00
+  message: "DEBUG - [main] BulkOpsClient.doImport(): Execution complete."
```
Comment (bfabio) [inline test/testdata/fixtures/logs.yml:132]:
```suggestion
  message: "[github.com/foobar/baz4] BAD publiccode.yml"
```

File: test/testdata/fixtures/logs.yml [hunk 9]
```diff
@@ -110,3 +110,28 @@
...
+
+- id: 5b45111e-042e-11ed-8d60-d8bbc146d165
+  created_at: 2010-08-01 23:59:59+00:00
+  updated_at: 2010-12-31 23:59:59+00:00
+  message: "DEBUG - [main] BulkOpsClient.sendValidationRequest(): Experienced SOAP Request Rate Limit error while sending the validation request. Will try to send again in 1 sec."
```
Comment (bfabio) [inline test/testdata/fixtures/logs.yml:137]:
```suggestion
  message: "[github.com/foobar/baz5] BAD publiccode.yml"
```

File: main_test.go [hunk 10]
```diff
@@ -3085,6 +3086,18 @@ func TestLogsEndpoints(t *testing.T) {
 				assert.Equal(t, "invalid date time format (RFC 3339 needed)", response["detail"])
 			},
 		},
+		{
+			description: `GET with "search" query param`,
+			query:       "GET /v1/logs?search=debug",
```
Comment (bfabio) [inline main_test.go:3091]: Adapting to fixtures

```suggestion
			query:       "GET /v1/logs?search=bad publiccode.yml",
```

**Pull request merged**